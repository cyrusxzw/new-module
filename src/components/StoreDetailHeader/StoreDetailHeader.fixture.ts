const StoreDetailHeaderFixture = {
  copy: {
    location: 'Location',
    email: 'Email',
    phone: 'Phone',
    openingHours: 'Opening hours',
  },
  email: 'consectetur.elit@aesop.com',
  location: 'Duis in massa sed, Suspendisse in massa sed hendrerit',
  phone: '+61 5555 1234',
  storeName: 'Lorem ipsum dolor',
};

export { StoreDetailHeaderFixture };
