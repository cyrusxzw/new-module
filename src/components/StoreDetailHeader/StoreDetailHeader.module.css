@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);
  border: 5px solid rgb(250, 38, 162);

  &,
  & * {
    box-sizing: border-box;
  }
}

.detailBlock {
  display: flex;
  flex-wrap: wrap;
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(15px);
  line-height: 1.4;
}

.detail {
  width: 100%;
  margin-bottom: 14px;

  @media (--viewport-md) {
    width: calc(100% * 1 / 2);
    padding: 0 40px 0 0;
  }

  @media (--viewport-xl) {
    width: calc(100% * 1 / 3);
  }
}

.detailHeading {
  margin-bottom: 4px;
  font-family: var(--font-family-sans-serif-medium);
}

.detailContent {
  white-space: pre-wrap;
  word-break: break-word;

  .dark & {
    color: var(--color-dark-label);
  }

  .light & {
    color: var(--color-light-label);
  }
}

.hyperlink {
  .dark & {
    color: var(--color-dark-label);
  }

  .light & {
    color: var(--color-light-label);
  }

  &:hover {
    .dark & {
      color: var(--color-dark-link);
    }

    .light & {
      color: var(--color-light-link);
    }
  }
}
