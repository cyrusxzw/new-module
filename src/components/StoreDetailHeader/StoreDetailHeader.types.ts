import type { ComponentWithoutChildren, Themes } from '~/types';
import type { HoursListItem } from '~/components/StoreHoursList/StoreHoursList.types';

type StoreDetailCopy = {
  location?: string;
  email?: string;
  phone?: string;
  facialAppointments?: string;
  facialAppointmentsLink?: string;
  openingHours?: string;
};

type StoreDetailHeaderProps = {
  alternateHoursNote?: string;
  className?: string;
  copy?: StoreDetailCopy;
  facialAppointments?: boolean;
  facialAppointmentsLink?: string;
  email?: string;
  location?: string;
  openingHours?: HoursListItem[];
  phone?: string;
  storeName: string;
  theme?: Themes;
};

type StoreDetailHeaderType = ComponentWithoutChildren<StoreDetailHeaderProps>;

export type { StoreDetailHeaderProps, StoreDetailHeaderType };
