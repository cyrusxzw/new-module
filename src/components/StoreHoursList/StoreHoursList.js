import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import find from 'lodash/find';
import { Heading } from '~/components/Heading/index.ts';
import styles from './StoreHoursList.module.css';

const StoreHoursList = ({
  alternateHoursNote,
  className,
  heading,
  hoursList,
  theme,
}) => {
  if (hoursList.length === 0) return null;

  const classSet = cx(styles.base, styles[theme], className);
  const hasAlternateHours = !!find(hoursList, 'isAlternate');

  return (
    <div className={classSet}>
      {heading && (
        <Heading
          className={styles.heading}
          level="4"
          size="xXSmall"
          theme={theme}
        >
          {heading}
        </Heading>
      )}

      <ul className={styles.list}>
        {hoursList.map((item) => (
          <li className={styles.item} key={item.id}>
            <span className={styles.dayName}>{item.dayName}</span>
            <span
              className={cx(styles.hours, {
                [styles.alternateHours]: item.isAlternate,
              })}
            >
              {item.hours}
            </span>
          </li>
        ))}
      </ul>
      {alternateHoursNote && hasAlternateHours && (
        <div className={styles.alternateHoursNote}>{alternateHoursNote}</div>
      )}
    </div>
  );
};

StoreHoursList.propTypes = {
  alternateHoursNote: PropTypes.string,
  className: PropTypes.string,
  heading: PropTypes.string,
  hoursList: PropTypes.arrayOf(
    PropTypes.shape({
      dayName: PropTypes.string,
      hours: PropTypes.string,
      id: PropTypes.string,
      isAlternate: PropTypes.bool,
    }),
  ),
  theme: PropTypes.oneOf(['dark', 'light']),
};

StoreHoursList.defaultProps = {
  alternateHoursNote: undefined,
  className: undefined,
  heading: undefined,
  hoursList: [],
  theme: 'dark',
};

export { StoreHoursList };
