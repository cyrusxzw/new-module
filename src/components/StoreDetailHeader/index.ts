import { StoreDetailHeader } from './StoreDetailHeader';

export { StoreDetailHeader };
