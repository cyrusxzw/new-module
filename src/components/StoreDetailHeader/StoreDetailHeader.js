import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { HEADING, HYPERLINK_STYLE_TYPES, GOOGLE_MAPS } from '~/constants';
import { Heading } from '~/components/Heading/index.ts';
import { Hyperlink } from '~/components/Hyperlink';
import { StoreHoursList } from '~/components/StoreHoursList/index.ts';
import { TwoColumnLayout } from '~/components/TwoColumnLayout/index.ts';
import styles from './StoreDetailHeader.module.css';

const DATA_TEST_REF_LOCATION = 'STORE_DETAILS_DIRECTION_LINK';
const DATA_TEST_REF_EMAIL = 'STORE_DETAILS_DIRECTION_EMAIL';
const DATA_TEST_REF_PHONE = 'STORE_DETAILS_DIRECTION_PHONE';

const StoreDetailHeader = ({
  alternateHoursNote,
  className,
  copy,
  email,
  location,
  openingHours,
  phone,
  storeName,
  theme,
}) => {
  const classSet = cx(styles.base, styles[theme], className);

  const contentBlocks = [
    {
      label: copy.location,
      content: location ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_LOCATION}
          hasTargetInNewWindow={true}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_TEXT_LINK}
          theme={theme}
          title={`${copy.location}: ${location}`}
          url={`${GOOGLE_MAPS.DIRECTIONS_URL_PREFIX}${location}`}
        >
          {location}
        </Hyperlink>
      ) : null,
      id: 'location',
    },
    {
      label: copy.email,
      content: email ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_EMAIL}
          hasTargetInNewWindow={true}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_TEXT_LINK}
          theme={theme}
          title={email}
          url={`mailto:${email}`}
        >
          {email}
        </Hyperlink>
      ) : null,
      id: 'email',
    },
    {
      label: copy.phone,
      content: phone ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_PHONE}
          hasTargetInNewWindow={false}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_NO_ICON_TEXT_LINK}
          theme={theme}
          title={`tel: ${phone}`}
          url={`tel:${phone}`}
        >
          {phone}
        </Hyperlink>
      ) : null,
      id: 'phone',
    },
    {
      label: copy.openingHours,
      content: (
        <StoreHoursList
          alternateHoursNote={alternateHoursNote}
          hoursList={openingHours}
          theme={theme}
        />
      ),
      id: 'openingHours',
    },
  ];

  return (
    <TwoColumnLayout
      className={classSet}
      content={
        <>
          <Heading
            level={HEADING.LEVEL.ONE}
            size={HEADING.SIZE.LARGE}
            theme={theme}
          >
            {storeName}
          </Heading>
          <div className={styles.detailBlock}>
            {contentBlocks
              .filter(({ label, content }) => label && content)
              .map(({ label, content, id }) => (
                <div className={styles.detail} key={id}>
                  <Heading
                    className={styles.detailHeading}
                    level={HEADING.LEVEL.THREE}
                    size={HEADING.SIZE.X_X_SMALL}
                    theme={theme}
                  >
                    {label}
                  </Heading>
                  <div className={cx(styles.detailContent)}>{content}</div>
                </div>
              ))}
          </div>
        </>
      }
      hasFullWidthContent={true}
      isReversed={true}
    />
  );
};

StoreDetailHeader.propTypes = {
  alternateHoursNote: PropTypes.string,
  className: PropTypes.string,
  copy: PropTypes.shape({
    location: PropTypes.string,
    email: PropTypes.string,
    phone: PropTypes.string,
    openingHours: PropTypes.string,
  }),
  email: PropTypes.string,
  location: PropTypes.string,
  openingHours: PropTypes.array,
  phone: PropTypes.string,
  storeName: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(['dark', 'light']),
};

StoreDetailHeader.defaultProps = {
  alternateHoursNote: undefined,
  className: undefined,
  copy: {
    location: undefined,
    email: undefined,
    phone: undefined,
    openingHours: undefined,
  },
  email: undefined,
  location: undefined,
  openingHours: undefined,
  phone: undefined,
  storeName: undefined,
  theme: 'dark',
};

export { StoreDetailHeader };
