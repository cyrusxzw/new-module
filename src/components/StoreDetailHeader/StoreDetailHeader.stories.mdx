import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { StorybookGridOverlay } from '~.storybook/components';
import { StoreHoursListFixture } from '~/components/StoreHoursList/StoreHoursList.fixture';
import { StoreDetailHeader } from './StoreDetailHeader';
import { StoreDetailHeaderFixture } from './StoreDetailHeader.fixture';

<Meta title="Components/StoreDetailHeader" component={StoreDetailHeader} />

# StoreDetailHeader

<Canvas>
  <Story name="Base component">
    {(() => {
      const copy = {
        location: knobs.text(
          'location',
          StoreDetailHeaderFixture.copy.location,
          'Copy',
        ),
        openingHours: knobs.text(
          'openingHours',
          StoreDetailHeaderFixture.copy.openingHours,
          'Copy',
        ),
        phone: knobs.text('phone', StoreDetailHeaderFixture.copy.phone, 'Copy'),
        email: knobs.text('email', StoreDetailHeaderFixture.copy.email, 'Copy'),
      };
      const email = knobs.text('email', StoreDetailHeaderFixture.email, 'Content');
      const location = knobs.text(
        'location',
        StoreDetailHeaderFixture.location,
        'Content',
      );
      const phone = knobs.text('phone', StoreDetailHeaderFixture.phone, 'Content');
      const storeName = knobs.text(
        'storeName',
        StoreDetailHeaderFixture.storeName,
        'Content',
      );
      const theme = knobs.select(
        'theme',
        ['dark', 'light'],
        'dark',
        'Presentation',
      );
      const implementation = {
        gridOverlay: {
          isActive: knobs.boolean('Show grid overlay', false, 'Implementation'),
          hasInvertedColours: knobs.boolean(
            'Invert grid overlay colours',
            false,
            'Implementation',
          ),
        },
      };
      return (
        <>
          <StoreDetailHeader
            alternateHoursNote={StoreHoursListFixture.alternateHoursNote}
            copy={copy}
            email={email}
            location={location}
            openingHours={StoreHoursListFixture.hoursList}
            phone={phone}
            storeName={storeName}
            theme={theme}
          />
          <StorybookGridOverlay
            isActive={implementation.gridOverlay.isActive}
            hasInvertedColours={implementation.gridOverlay.hasInvertedColours}
          />
        </>
      );
    })()}
  </Story>
</Canvas>

## API

<ArgsTable of={StoreDetailHeader} />
