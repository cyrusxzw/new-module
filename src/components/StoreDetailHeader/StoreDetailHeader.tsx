import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { HYPERLINK_STYLE_TYPES, GOOGLE_MAPS } from '~/constants';
import { Heading } from '~/components/Heading';
import { Hyperlink } from '~/components/Hyperlink';
import { StoreHoursList } from '~/components/StoreHoursList';
import { TwoColumnLayout } from '~/components/TwoColumnLayout';
import type { StoreDetailHeaderType } from './StoreDetailHeader.types';
import styles from './StoreDetailHeader.module.css';

const DATA_TEST_REF_LOCATION = 'STORE_DETAILS_DIRECTION_LINK';
const DATA_TEST_REF_EMAIL = 'STORE_DETAILS_DIRECTION_EMAIL';
const DATA_TEST_REF_PHONE = 'STORE_DETAILS_DIRECTION_PHONE';
const DATA_TEST_REF_FACIALAPPOINTMENTS = 'STORE_DETAILS_DIRECTION_FACIALAPPOINTMENTS';

const StoreDetailHeader: StoreDetailHeaderType = ({
  alternateHoursNote,
  className,
  copy,
  facialAppointments,
  facialAppointmentsLink,
  email,
  location,
  openingHours,
  phone,
  storeName,
  theme,
}) => {
  const currentTheme = useThemeContext(theme, 'dark');
  const classSet = cx(styles.base, styles[currentTheme], className);
  const wrapperClass = cx(styles.wrapper, className);

  console.log('-copy: ', copy);

  const contentBlocks = [
    {
      label: copy?.location,
      content: location ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_LOCATION}
          hasTargetInNewWindow={true}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_TEXT_LINK}
          theme={currentTheme}
          title={`${copy?.location}: ${location}`}
          url={`${GOOGLE_MAPS.DIRECTIONS_URL_PREFIX}${location}`}
        >
          {location}
        </Hyperlink>
      ) : null,
      id: 'location',
    },
    {
      label: copy?.email,
      content: email ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_EMAIL}
          hasTargetInNewWindow={true}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_TEXT_LINK}
          theme={currentTheme}
          title={`${copy?.email}: ${email}`}
          url={`mailto:${email}`}
        >
          {email}
        </Hyperlink>
      ) : null,
      id: 'email',
    },
    {
      label: copy?.phone,
      content: phone ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_PHONE}
          hasTargetInNewWindow={false}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_NO_ICON_TEXT_LINK}
          theme={currentTheme}
          title={`${copy?.phone}: ${phone}`}
          url={`tel:${phone}`}
        >
          {phone}
        </Hyperlink>
      ) : null,
      id: 'phone',
    },
    {
      label: copy?.facialAppointments,
      content: facialAppointments ? (
        <Hyperlink
          className={styles.hyperlink}
          dataTestRef={DATA_TEST_REF_FACIALAPPOINTMENTS}
          hasTargetInNewWindow={true}
          style={HYPERLINK_STYLE_TYPES.EXTERNAL_TEXT_LINK}
          theme={currentTheme}
          title={`${copy?.facialAppointmentsLink}`}
          url={facialAppointmentsLink}
        >
          {facialAppointments}
        </Hyperlink>
      ) : null,
      id: 'facialAppointments',
    },
    {
      label: copy?.openingHours,
      content: (
        <StoreHoursList
          alternateHoursNote={alternateHoursNote}
          hoursList={openingHours}
          theme={currentTheme}
        />
      ),
      id: 'openingHours',
    },
  ];

  console.log('-contentBlocks: ', contentBlocks);

  return (
    <TwoColumnLayout
      className={classSet}
      content={
        <div className={wrapperClass}>
          <Heading level="1" size="xLarge" theme={currentTheme}>
            {storeName}
          </Heading>
          <div className={styles.detailBlock}>
            {contentBlocks
              .filter(({ label, content }) => label && content)
              .map(({ label, content, id }) => (
                <div className={styles.detail} key={id}>
                  <Heading
                    className={styles.detailHeading}
                    level="3"
                    size="xXSmall"
                    theme={currentTheme}
                  >
                    {label}
                  </Heading>
                  <div className={cx(styles.detailContent)}>{content}</div>
                </div>
              ))}
          </div>
        </div>
      }
      hasFullWidthContent={true}
      isReversed={true}
    />
  );
};

export { StoreDetailHeader };
