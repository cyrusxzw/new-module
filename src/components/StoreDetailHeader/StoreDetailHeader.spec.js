import React from 'react';
import { render } from '@testing-library/react';
import { StoreHoursListFixture } from '~/components/StoreHoursList/StoreHoursList.fixture';
import { StoreDetailHeader } from './StoreDetailHeader';
import { StoreDetailHeaderFixture } from './StoreDetailHeader.fixture';

describe('<StoreDetailHeader />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <StoreDetailHeader
        alternateHoursNote={StoreHoursListFixture.alternateHoursNote}
        copy={{
          location: StoreDetailHeaderFixture.copy.location,
          openingHours: StoreDetailHeaderFixture.copy.openingHours,
          phone: StoreDetailHeaderFixture.copy.phone,
          email: StoreDetailHeaderFixture.copy.email,
        }}
        email={StoreDetailHeaderFixture.email}
        location={StoreDetailHeaderFixture.location}
        openingHours={StoreHoursListFixture.hoursList}
        phone={StoreDetailHeaderFixture.phone}
        storeName={StoreDetailHeaderFixture.storeName}
        theme={'dark'}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
