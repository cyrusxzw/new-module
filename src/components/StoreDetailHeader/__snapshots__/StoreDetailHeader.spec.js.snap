// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreDetailHeader /> should render base component correctly 1`] = `
<div>
  <div
    class="base reverse contentOnly dark base dark"
  >
    <div
      class="content fullWidth"
      id="mainColumn"
    >
      <h1
        class="base large dark"
      >
        Lorem ipsum dolor
      </h1>
      <div
        class="detailBlock"
      >
        <div
          class="detail"
        >
          <h3
            class="base xXSmall dark detailHeading"
          >
            Location
          </h3>
          <div
            class="detailContent"
          >
            <a
              class="base hasIcon hasExternalIcon left dark hyperlink"
              data-test-ref="STORE_DETAILS_DIRECTION_LINK"
              href="https://www.google.com/maps?saddr=Current+Location&daddr=Duis in massa sed, Suspendisse hendrerit"
              target="_blank"
              title="Location: Duis in massa sed, Suspendisse hendrerit"
            >
              Duis in massa sed, Suspendisse hendrerit
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-1"
                  class="base rightUpArrow dark"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-1"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </div>
        </div>
        <div
          class="detail"
        >
          <h3
            class="base xXSmall dark detailHeading"
          >
            Email
          </h3>
          <div
            class="detailContent"
          >
            <a
              class="base hasIcon hasExternalIcon left dark hyperlink"
              data-test-ref="STORE_DETAILS_DIRECTION_EMAIL"
              href="mailto:consectetur.elit@aesop.com"
              target="_blank"
              title="consectetur.elit@aesop.com"
            >
              consectetur.elit@aesop.com
               
              <i
                aria-hidden="true"
                class="icon"
              >
                <svg
                  aria-labelledby="rightUpArrow-2"
                  class="base rightUpArrow dark"
                  data-testid="data-testid-Icon"
                  focusable="false"
                  height="12"
                  role="img"
                  style="height: 12px; width: 12px;"
                  tabindex="-1"
                  viewBox="0 0 50 50"
                  width="12"
                >
                  <title
                    id="rightUpArrow-2"
                  >
                    rightUpArrow
                  </title>
                  <g>
                    <path
                      d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                    />
                  </g>
                </svg>
              </i>
            </a>
          </div>
        </div>
        <div
          class="detail"
        >
          <h3
            class="base xXSmall dark detailHeading"
          >
            Phone
          </h3>
          <div
            class="detailContent"
          >
            <a
              class="base left dark hyperlink"
              data-test-ref="STORE_DETAILS_DIRECTION_PHONE"
              href="tel:+61 5555 1234"
              target="_self"
              title="tel: +61 5555 1234"
            >
              +61 5555 1234
            </a>
          </div>
        </div>
        <div
          class="detail"
        >
          <h3
            class="base xXSmall dark detailHeading"
          >
            Opening hours
          </h3>
          <div
            class="detailContent"
          >
            <div
              class="base dark"
            >
              <ul
                class="list"
              >
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Monday
                  </span>
                  <span
                    class="hours"
                  >
                    11am - 4pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Tuesday
                  </span>
                  <span
                    class="hours alternateHours"
                  >
                    10am - 3pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Wednesday
                  </span>
                  <span
                    class="hours"
                  >
                    10am - 3pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Thursday
                  </span>
                  <span
                    class="hours"
                  >
                    10am - 3pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Friday
                  </span>
                  <span
                    class="hours"
                  >
                    10am - 3pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Saturday
                  </span>
                  <span
                    class="hours alternateHours"
                  >
                    10am - 3pm
                  </span>
                </li>
                <li
                  class="item"
                >
                  <span
                    class="dayName"
                  >
                    Sunday
                  </span>
                  <span
                    class="hours"
                  >
                    10am - 3pm
                  </span>
                </li>
              </ul>
              <div
                class="alternateHoursNote"
              >
                Special opening hours
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
